### PR TITLE
[Merged by Bors] - TO-3041 integration test for migration

### PR DESCRIPTION
**Pull request by rustonaut**
**based on:**

- #600
- #601 
- #602 
- #607

### CODE REVIEW

File: discovery_engine/lib/src/domain/repository/active_document_repo.dart [hunk 1]
```diff
@@ -34,4 +34,10 @@ abstract class ActiveDocumentDataRepository {
 
   /// Remove data associated with the given active documents.
   Future<void> removeByIds(Iterable<DocumentId> ids);
+
+  /// Clears the repository.
+  Future<void> clear();
+
+  /// Checks if the repository is semantically empty.
```
Comment (biernacki) [inline discovery_engine/lib/src/domain/repository/active_document_repo.dart:41]: 🎨 
```suggestion
  /// Indicates if the repository is semantically empty.
```

File: discovery_engine/test/integration/migration_test.dart [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        expect(repos.activeDocumentDataRepository.isEmpty, isTrue);
+        expect(repos.sourceReactedRepository.isEmpty, isTrue);
+        expect(repos.sourcePreferenceRepository.isEmpty, isTrue);
+      });
+    });
+
+    test('works with mostly empty db (b)', () async {
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        await repos.documentRepository.update(docs[0]);
```
Comment (biernacki) [inline discovery_engine/test/integration/migration_test.dart:239]: 🎨 a bit nicer to read, but just a suggestion
```suggestion
        await repos.documentRepository.update(docs.first);
```

File: discovery_engine/test/integration/migration_test.dart [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+        expect(repos.activeDocumentDataRepository.isEmpty, isTrue);
+        expect(repos.sourceReactedRepository.isEmpty, isTrue);
+        expect(repos.sourcePreferenceRepository.isEmpty, isTrue);
+      });
+    });
+
+    test('works with needing migration for all data', () async {
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        //FIXME test engine state migration by having a known valid engine state blob
```
Comment (biernacki) [inline discovery_engine/test/integration/migration_test.dart:263]: that's a good idea, I think I have some hive files with some engine state that we could use from the datetime type migration issue, or if you need a particular state, then I can help also to create it

**Review by biernacki — APPROVED** [review]
just minor comments, but great work otherwise

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**